### ISSUE TITLE
Discuss `zcap.insert` validation.

### ISSUE THREAD

**Issue by mattcollier** [digitalbazaar/bedrock-zcap-storage-http#5]
@dlongley here's the code referenced in the TODO: https://github.com/digitalbazaar/bedrock-zcap-storage/blob/master/lib/storage.js#L240-L245

Does a ZCAP really need a controller AND an invoker?

_Originally posted by @mattcollier in https://github.com/digitalbazaar/bedrock-zcap-storage-http/pull/4_

**Comment by dlongley** [comment]
The `controller` there is the party in charge of storing the zcap for later use -- which may not be the same party as the one invoking it. Consider a person with an account on a system -- that has N many profile agents. Consider a use case where that person wants to list all of the zcaps they have associated with their account. We could consider redesigning the storage here provided that use cases such as the one above can still be supported.

**Comment by mattcollier** [comment]
@dlongley you said it "may not be the same party".  Are our current use cases such that they are always different parties anyway?  I'm trying to understand why the model here is any different than in ocap-ld.

**Comment by dlongley** [comment]
The `controller` here is not `capability.controller`. It's whoever is in control of *storing* (or *deleting*) the capability.

**Comment by mattcollier** [comment]
is the `invoker` the `capability.invoker`?

**Comment by dlongley** [comment]
Yes, `invoker` is `capability.invoker`.

**Comment by mattcollier** [comment]
Right, OK, I think we've identified the source of confusion.  :)

**Comment by dlongley** [comment]
`controller` is the controller of the record that contains the zcap (they can store/delete the record), which is not necessarily the same party as the one that controls the zcap (they can invoke/delegate the zcap). How should we remedy the confusion?

**Comment by mattcollier** [comment]
@dlongley putting `controller` parameter to the API aside for the moment.  

This API requires that ALL zcaps inserted here will have an `invoker` property.  As I understand it, that is only sometimes a requirement because there are use cases where having only `controller` in the `capability` would be correct and adequate.  Is that correct.

**Comment by dlongley** [comment]
@mattcollier, yes, that's correct. We need to fix the code to support accepting `capability.invoker` or `capability.controller` and using  both in the index.
